monadic-rx/shared/src/main/scala/mhtml/rx.scala

```diff
@@ -237,13 +237,36 @@ object Rx {
 }
 
 /** A smart variable that can be set manually. */
-class Var[A](initialValue: Option[A], register: Var[A] => Cancelable) extends Rx[A] {
+class Var[A](
+  initialValue: Option[A], register: Var[A] => Cancelable,
+  allowMultiMutations: Boolean = false
+) extends Rx[A] {
   // Last computed value, retained to be sent to new subscribers as they come in.
   private[mhtml] var cacheElem: Option[A] = initialValue
   // Current registration to the feeding `Rx`, canceled whenever nobody's listening.
   private[mhtml] var registration: Cancelable = Cancelable.empty
   // Mutable set of all currently subscribed functions, implementing with an `Array`.
   private[mhtml] val subscribers = buffer.empty[A => Unit]
+  // Keeps track of mutable call locations
+  private[mhtml] var mutationLocations: Set[Int] = Set()
+  private[mhtml] def wasSet: Boolean = mutationLocations.size > 1
+
+  private[mhtml] def addUpdateCall(): Unit = {
+    if (!Var.isProduction) {
+      if (!allowMultiMutations) {
+        // Avoid doing expensive safety checks in optimized code that
+        // should be caught during testing.
+        val straceSeq = new RuntimeException().getStackTrace.toSeq
+        val lineNumHash = straceSeq.map(stl => stl.getLineNumber).mkString("\n").hashCode.toString
+        val contentHash = straceSeq.map(
+          stl => stl.getFileName + stl.getClassName + stl.getClassName
+        ).mkString("\n").hashCode.toString
```

Comment:
It might be interesting to see if we could keep more info about this stack-trace to say something like "error: var was mutate at two locations, the first time from <file:line>, the second time from <file:line>.